validator/client/src/main/java/tech/pegasys/teku/validator/client/ActiveKeyManager.java

```diff
@@ -160,6 +165,33 @@ public DeleteRemoteKeysResponse deleteExternalValidators(List<BLSPublicKey> vali
     return new DeleteRemoteKeysResponse(deletionResults);
   }
 
+  private void deleteExternalDoppelgangers(final Set<BLSPublicKey> publicKeys) {
+    for (BLSPublicKey publicKey : publicKeys) {
+      Optional<Validator> maybeOwnedValidator =
+          validatorLoader.getOwnedValidators().getValidator(publicKey);
+      if (maybeOwnedValidator.isPresent()) {
+        if (maybeOwnedValidator.get().isReadOnly()) {
+          LOG.warn("Unable remove read only validator doppelganger.");
+          continue;
+        }
+        final Signer signer = maybeOwnedValidator.get().getSigner();
+        signer.delete();
+        DeleteKeyResult deleteKeyResult = validatorLoader.deleteExternalMutableValidator(publicKey);
+        if (deleteKeyResult.equals(DeleteKeyResult.success())) {
+          validatorLoader.getOwnedValidators().removeValidator(publicKey);
+          LOG.info(
+              "Removed validator doppelgangers public key {}", publicKey.toAbbreviatedString());
+
+        } else {
+          LOG.warn(
+              String.format(
+                  "Unable to remove validator doppelganger public key %s. %s",
+                  publicKey.toAbbreviatedString(), deleteKeyResult.getMessage().orElse("")));
```

Comment:
I'd say this would be an error not a warning.  But under what situation could we be failing to remove a validator? Does it potentially mean that we are left in an unsafe situation with the validator still active and potentially getting slashed?

Reply:
Theoretically this shouldn't happen when importing new keys. The deletion result is more relevant when it comes to existing keys deletion. I added this log for debug purposes (If for some reasons the deletion fails). And yes if it happens it means that the validator will still stay active and hence a potential slashing (which is the downside of not shutting down the VC and rather trying to remove the keys).
I changed it to an error anyway.